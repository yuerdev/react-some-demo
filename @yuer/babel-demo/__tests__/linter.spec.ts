// import { transformSync } from "@babel/core";
// import { dirname } from "path";
// import { fileURLToPath } from "url";
// import Linter from "../linter";

// const __dirname = dirname(fileURLToPath(import.meta.url));

// const compileToSnapshot = (code: string) => {
//     const t = transformSync(code, {
//         root: __dirname,
//         parserOpts: {
//             plugins: ["jsx"],
//         },
//         plugins: [[Linter,{
//             fix: true
//         }]],
//     });

//     return `
// ${code}
    
// ↓ ↓ ↓ ↓ ↓ ↓

// ${t?.code}  
// `;
// };

// describe("for direction", () => {
//     it("should direction", () => {
//         const result = compileToSnapshot(`
//         for (var i = 0; i < 10; i++) {
//         }
        
//         for (var i = 10; i >= 0; i--) {
//         }
//         for (var i = 0; i < 10; i--) {
//         }
        
//         for (var i = 10; i >= 0; i++) {
//         }
//     `);

//         expect(result).toMatchSnapshot();
//     });
    
//     it(" ==", () => {
//         const result = compileToSnapshot(`
//         a == b
//         foo == true
//         bananas != 1
//         value == undefined
//         typeof foo == 'undefined'
//         'hello' != 'world'
//         0 == 0
//         true == true
//     `);

//         expect(result).toMatchSnapshot();
//     });
// });