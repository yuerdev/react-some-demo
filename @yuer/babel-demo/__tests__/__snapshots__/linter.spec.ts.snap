// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`for direction  == 1`] = `
"

        a == b
        foo == true
        bananas != 1
        value == undefined
        typeof foo == 'undefined'
        'hello' != 'world'
        0 == 0
        true == true
    
    
↓ ↓ ↓ ↓ ↓ ↓

a === b;
foo === true;
bananas !== 1;
value === undefined;
typeof foo === 'undefined';
'hello' != 'world';
0 == 0;
true == true;  
"
`;

exports[`for direction should direction 1`] = `
"

        for (var i = 0; i < 10; i++) {
        }
        
        for (var i = 10; i >= 0; i--) {
        }
        for (var i = 0; i < 10; i--) {
        }
        
        for (var i = 10; i >= 0; i++) {
        }
    
    
↓ ↓ ↓ ↓ ↓ ↓

for (var i = 0; i < 10; i++) {}

for (var i = 10; i >= 0; i--) {}

for (var i = 0; i < 10; i--) {}

for (var i = 10; i >= 0; i++) {}  
"
`;
